compiler/AST/ModuleSymbol.cpp

```diff
@@ -424,8 +422,33 @@ void ModuleSymbol::printTableOfContents(std::ostream* file) {
 /*
  * Returns name of module, including any prefixes that have been set.
  */
-std::string ModuleSymbol::docsName() {
-  return this->name;
+std::string ModuleSymbol::docsName() const {
+  return name;
+}
+
+/*
+ * Generate a name that represents the path to the module
+ * For a top-level module this is simply the name.
+ * For nested modules that name corresponds to the "use name"
+ */
+
+std::string ModuleSymbol::path() const {
```

Comment:
I suspect that chpldoc might do something similar when generating the name for a module.  Could it make use of this new function?  (not a priority, just a musing)